src/test/FuzzerImpl.cpp

```diff
@@ -235,6 +329,17 @@ struct xdr_fuzzer_compactor
         auto b = stellar::FuzzUtils::getShortKey(ed25519);
         put_bytes(&b, 1);
     }
+
+    template <typename T>
+    typename std::enable_if<std::is_same<stellar::LedgerKey, T>::value>::type
+    operator()(T const& key)
+    {
+        // Convert LedgerKey to 1 byte for indexing into an array of LedgerKeys
+        // that have been mentioned in the XDR of fuzzer operations.
+        check(1);
+        auto b = stellar::FuzzUtils::getOrCreateLedgerKeyID(key);
```

Comment:
Are you suggesting not using a counter at all, but indexing into a static array of `LedgerKey`s using a single byte from the `LedgerKey`?  If so, which byte -- would that be key-type-dependent to make sure that it was part of something pseudorandom such as a hash or public key?  And wouldn't that require keeping track of which individual entries in the array contain valid `LedgerKey`s?  That all sounds much more complicated than a counter, so I'm pretty sure I don't understand what you mean.

Reply:
Yes, that's what I'm suggesting. It would depend on the key type, but this is a simple switch statement.

But no, I do not believe it requires keeping track of anything (which was my main justification for saying that this is too complicated). Let me give you three arguments:
- This code is in the compactor, which is only used in `xdr_to_fuzzer_opaque` which in turn is only used by the two implementations of `genFuzz`. At this stage you don't know anything about the set of valid ledger keys, because the fuzzer isn't even initialized (check what `runGenFuzz` does).  
- This code is used to generate the initial fuzzer corpus. The fuzzer is going to randomly mutate that input anyway, so clearly the fuzzer can generate inputs that don't map to any known ledger key. But then you shouldn't worry about mapping to any known ledger key either.
- The unpacker is figuring out how to map the compacted ledger keys to actual (or maybe not actual!) ledger keys. If it is being figured out there, why does it also need to be figured out here?

Finally, what you're doing is not just a counter--`getOrCreateLedgerKeyID` is a 25 line function with two branches and several calls into the standard library.

Reply:
Upon re-reading your question, I'm not sure I answered it clear so let me clarify. I _am_ suggesting not to use a counter, but I _am not_ "suggesting to index into a static array of `LedgerKeys` using a single byte from the `LedgerKey`". All I'm suggesting is to just grab a single pseudorandom byte from the input and output that, which is what `getShortKey` does as I mentioned in the original comment. There isn't any static array to index into.

Reply:
I have tried to implement your intent.